src/pveclib/vec_int64_ppc.h

```diff
@@ -3028,6 +3073,43 @@ vec_srdi (vui64_t vra, const unsigned int shb)
       /* Vector Shift right bytes based on the lower 6-bits of
          corresponding element of rshift.  */
       result = vec_vsrd (vra, rshift);
+#else
+      /*
+       * POWER7 and earlier do not have vsrd. So use the vector shift
+       * right bit/octet instructions. But these may shift bits from
+       * element 0 in the high bits of element 1. So generate a mask of
+       * '1's, shifted right by the same shb and rotated into the
+       * element 1 position.
+       */
+      vui8_t rshift;
+
+      if (__builtin_constant_p (shb) && (shb < 16))
+	rshift = vec_splat_u8(shb);
+      else
+	rshift = vec_splats ((unsigned char) shb);
+
+      {
+	vui8_t sr_a;
+	vui8_t sr_m = (vui8_t) vec_splat_s8(-1);
+
+	sr_a = ((vui8_t) vra);
+	if (shb > 7)
+	  {
+	    /* Vector Shift Left By Octet by bits 121-124 of lshift.  */
+	    sr_m = vec_sro (sr_m, rshift);
+	    sr_a = vec_sro ((vui8_t) vra, rshift);
+	  }
+	if ((shb & 7) != 0)
+	  {
+	    /* Vector Shift Left by bits 125-127 of lshift.  */
```

Comment:
Likewise here.

Reply:
Yup a pasto from vec_sldi. Fixed in update.